src/Stack/ModuleInterface.hs

```diff
@@ -0,0 +1,425 @@
+module Stack.ModuleInterface
```

Comment:
General question: is this code copied directly from somewhere else?

Reply:
It's very bad for me to hear that... I've written all the code, which include reversing the different protocol versions.

Reply:
I didn't mean to say anything bad, sorry! I was asking because of things like the `FastString` newtype.

Reply:
Ohh, it's okay, I thought that you were looking for plagiarism.. The names are indeed inspired by the GHC code (mnemonic for me while I was coding)